### ISSUE TITLE
Compile failed, "module Mongo.Pool is not loaded and could not be found"

### ISSUE THREAD

**Issue by vfsoraki** [michalmuskala/mongodb_ecto#96]
I tried installing mongodb_ecto, here is the relevant `mix.exs`:

```
# Configuration for the OTP application                                     
#                                                                           
# Type "mix help compile.app" for more information                          
def application do
  [applications: [:logger, :mongodb_ecto, :ecto],
   mod: {MyApp, []}]
end

# Dependencies can be Hex packages:                                         
#                                                                           
#   {:mydep, "~> 0.3.0"}                                                    
#                                                                           
# Or git/path repositories:                                                 
#                                                                           
#   {:mydep, git: "https://github.com/elixir-lang/mydep.git", tag: "0.1.0"} 
#                                                                           
# Type "mix help deps" for more examples and options                        
defp deps do
  [{:mongodb_ecto, "~> 0.1"}]
end
```

Then I ran `mix deps.get` and all was fine.

Running `mix compile` yields this error

```
== Compilation error on file lib/mongo_ecto/admin_pool.ex ==
** (CompileError) lib/mongo_ecto/admin_pool.ex:4: module Mongo.Pool is not loaded and could not be found
    (elixir) expanding macro: Kernel.use/2
    lib/mongo_ecto/admin_pool.ex:4: Mongo.Ecto.AdminPool (module)
    (elixir) lib/kernel/parallel_compiler.ex:116: anonymous fn/4 in Kernel.ParallelCompiler.spawn_compilers/1

could not compile dependency :mongodb_ecto, "mix compile" failed. You can recompile this dependency with "mix deps.compile mongodb_ecto", update it with "mix deps.update mongodb_ecto" or clean it with "mix deps.clean mongodb_ecto"
```

I googled, but nothing showed up. Since ecto is compiling fine, I suppose there is a problem with mongodb_ecto. Here is the content of `mix.lock` to see the versions:

```
%{"connection": {:hex, :connection, "1.0.4", "a1cae72211f0eef17705aaededacac3eb30e6625b04a6117c1b2db6ace7d5976", [:mix], []},
  "db_connection": {:hex, :db_connection, "1.1.0", "b2b88db6d7d12f99997b584d09fad98e560b817a20dab6a526830e339f54cdb3", [:mix], [{:connection, "~> 1.0.2", [hex: :connection, optional: false]}, {:poolboy, "~> 1.5", [hex: :poolboy, optional: true]}, {:sbroker, "~> 1.0", [hex: :sbroker, optional: true]}]},
  "decimal": {:hex, :decimal, "1.3.1", "157b3cedb2bfcb5359372a7766dd7a41091ad34578296e951f58a946fcab49c6", [:mix], []},
  "ecto": {:hex, :ecto, "1.0.7", "d8deca5d5a03e7c9ed4821e6f1f85dad736aca960785f154e3a8c1bee1aeaa13", [:mix], [{:decimal, "~> 1.0", [hex: :decimal, optional: false]}, {:mariaex, "~> 0.4.1", [hex: :mariaex, optional: true]}, {:poison, "~> 1.0", [hex: :poison, optional: true]}, {:poolboy, "~> 1.4", [hex: :poolboy, optional: false]}, {:postgrex, "~> 0.9.1", [hex: :postgrex, optional: true]}, {:sbroker, "~> 0.7", [hex: :sbroker, optional: true]}]},
  "mongodb": {:hex, :mongodb, "0.2.0", "95f85074534bb0a65c4983995903a458e36d0b096f62c1bbdd56bc196f48cf80", [:mix], [{:connection, "~> 1.0", [hex: :connection, optional: false]}, {:db_connection, "~> 1.0", [hex: :db_connection, optional: false]}]},
  "mongodb_ecto": {:hex, :mongodb_ecto, "0.1.4", "d8eebaed05851802b0e7cd53dcfbb4b62a5819e8acd7c584b8eaa3c8ead43c2c", [:mix], [{:ecto, "~> 1.0.0", [hex: :ecto, optional: false]}, {:mongodb, "~> 0.1", [hex: :mongodb, optional: false]}]},
  "poolboy": {:hex, :poolboy, "1.5.1", "6b46163901cfd0a1b43d692657ed9d7e599853b3b21b95ae5ae0a777cf9b6ca8", [:rebar], []}}
```

I see nothing wrong in there.

And here is elixir version:

```
Erlang/OTP 19 [erts-8.0.3] [source] [64-bit] [smp:4:4] [async-threads:10] [hipe] [kernel-poll:false]

Elixir 1.3.2
```

Since I found nothing by googling, I assume I am doing something wrong, though it seems not. I would add bug label if it shows up to be a problem with mongodb_ecto. I already googled to find what is `Mongo.Pool` but again nothing showed up. I would be more than happy to fix this problem, if I had a clue on the cause.

**Comment by vfsoraki** [comment]
I think I have found the problem. This line in `mix.lock`

```
  "mongodb": {:hex, :mongodb, "0.2.0", "95f85074534bb0a65c4983995903a458e36d0b096f62c1bbdd56bc196f48cf80", [:mix], [{:connection, "~> 1.0", [hex: :connection, optional: false]}, {:db_connection, "~> 1.0", [hex: :db_connection, optional: false]}]},
```

caused the problem. Adding `{:mongodb, "~> 0.1.1"}` to `mix.exs` fixed the problem.

I think the dependency of `mongodb_ecto` should be changed to `~> 0.1.0`, since it won't work with `0.2` branch of `mongodb`.

**Comment by vfsoraki** [comment]
Seems to be fixed in master, commit 6d47b6ee3c7fb6a705a298eb286de97f647cc970, though no notification here. I was just going to create PR that I saw it was fixed.

Thanks anyway, I'm closing the issue.

**Comment by michalmuskala** [comment]
I've just released 0.1.5, which has a stricter dependency on mongodb, and should fix this issue.
